libcxx/test/benchmarks/ContainerBenchmarks.h

```diff
@@ -48,6 +48,76 @@ void BM_Assignment(benchmark::State& st, Container) {
   }
 }
 
+// Wrap any Iterator into an input iterator
+template <typename Iterator>
+class InputIterator {
```

Comment:
Why is this class required for the benchmark below? Why not simply pass `c.assign(in.begin(), in.end())`? Is it because otherwise the `__assign_with_size()` overload is used?

Reply:
Exactly. This `InputIterator` wrapper class ensures that `vector::assign(_InputIterator, _InputIterator)` invokes `__assign_with_sentinel`, not `__assign_with_size`.